x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Analyzer.java

```diff
@@ -929,15 +929,15 @@ public LogicalPlan apply(LogicalPlan plan, AnalyzerContext context) {
         }
 
         private static Expression cast(ScalarFunction f, EsqlFunctionRegistry registry) {
+            if (f instanceof In in) {
+                return processIn(in);
+            }
             if (f instanceof EsqlScalarFunction esf) {
                 return processScalarFunction(esf, registry);
             }
             if (f instanceof EsqlArithmeticOperation || f instanceof BinaryComparison) {
                 return processBinaryOperator((BinaryOperator) f);
             }
-            if (f instanceof In in) {
-                return processIn(in);
-            }
```

Comment:
Can we use `processScalarFunction` for this one or is it incompatible somehow?

Reply:
> Can we use `processScalarFunction` for this one or is it incompatible somehow?

This is a good point! It is not quite compatible for now, one reason is that In is not a registered function yet, it is under operator/comparison, there will be a little more work to make it a registered function. I'll look into it, and if possible, create a follow up PR for it.